### PR TITLE
:construction_worker: Add Github Action to build and deploy dev image when merged to master

### PR DESCRIPTION
**Pull request by haynescd**
tag name will be `post-release-(current-tag)-github-hash`

Closes [AD-2329](https://d3b.atlassian.net/browse/AD-2329)

[AD-2329]: https://d3b.atlassian.net/browse/AD-2329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Build and Publish Image"
+        run: |
+          docker build . --tag ghcr.io/kids-first/kf-api-dataservice/kf-api-dataservice:post-release-${ git tag --sort=taggerdate | tail -1 | head -1 }-${GITHUB_SHA:0:7}
```
Comment (znatty22) [inline .github/workflows/deploy.yml]: So this gives us the latest-release tag which we also apply to the latest release. Also for some reason git tag --sort=taggerdate doesn't sort by date
```
git tag --sort=taggerdate | tail -1 | head -1 
```

This seems to work: 
```
git tag --sort=committerdate | grep -E '[0-9]' | tail -1 | cut -b 1-7
```
It'll give us 1.32.0 right now. Modified it from https://gist.github.com/rponte/fdc0724dd984088606b0?permalink_comment_id=3180306#gistcomment-3180306

**Review by znatty22 — COMMENTED** [review]
Just putting a note here to remind us that we need to wait until https://github.com/kids-first/kf-api-dataservice/pull/684 merges, so that when this action runs itll build the image w the dev modifications

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (haynescd) [reply .github/workflows/deploy.yml]: will update

**Review by znatty22 — APPROVED** [review]

**Review by devbyaccident — APPROVED** [review]
lgtm. If there's any issues after merge, we can handle it after.

**Pull request merged**